accounts/models.py

```diff
@@ -43,6 +44,10 @@ def is_student(self):
     def has_name(self):
         return self.first_name is not None and self.first_name != ""
 
+    @property
+    def has_nickname(self):
+        return self.nickname is not None and self.nickname != ""
```

Comment:
I feel like a better solution is to just not let them set an empty string nickname instead of guarding against this case in so many places